# HG changeset patch
# User alta88@fixall.com
# Date 1748459818 21600
# Parent  c00f715259aa67c425d457cbfcb30a05983ba4dd
Feed Subscribe dialog enhancements
- Fix links; show urls.

diff --git a/mail/themes/shared/mail/feedSubscribe.css b/mail/themes/shared/mail/feedSubscribe.css
--- a/mail/themes/shared/mail/feedSubscribe.css
+++ b/mail/themes/shared/mail/feedSubscribe.css
@@ -17,8 +17,19 @@
     &:not([lwtheme]) {
       --foldertree-background: -moz-Dialog;
     }
   }
+  :any-link, .text-link {
+    &:not(:hover) {
+      text-decoration: none;
+    }
+  }
+
+  @media -moz-pref("layout.css.always_underline_links") {
+    .text-link {
+      text-decoration: underline !important;
+    }
+  }
 }
 
 html,
 body {
@@ -42,8 +53,9 @@ body {
 }
 
 #learnMore {
   align-self: end;
+  margin-block-end: 2px;
 }
 
 /* Tree styling - match folderPane */
 #folderTree {
@@ -219,8 +231,12 @@ label:has(:disabled) {
   text-overflow: ellipsis;
   flex: 1;
 }
 
+#statusBox {
+  height: 3em;
+}
+
 #locationBox,
 #folderBox,
 #statusBox,
 #actionButtonsBox {
diff --git a/mailnews/extensions/newsblog/feed-subscriptions.js b/mailnews/extensions/newsblog/feed-subscriptions.js
--- a/mailnews/extensions/newsblog/feed-subscriptions.js
+++ b/mailnews/extensions/newsblog/feed-subscriptions.js
@@ -101,8 +101,16 @@ var FeedSubscriptions = {
 
     // For observe notices.
     Services.ww.registerNotification(this);
 
+    const learnMoreLink = document.getElementById("learnMoreLink");
+    learnMoreLink.href = Services.prefs.getCharPref("rss.learnmore.url");
+    learnMoreLink.addEventListener("mouseenter", this.onMouseOverLink);
+
+    const locationValidate = document.getElementById("locationValidate");
+    locationValidate.href = Services.prefs.getCharPref("rss.validator.url");
+    locationValidate.addEventListener("mouseenter", this.onMouseOverLink);
+
     // Set up some strings.
     document.getElementById("createSubFolderLabel").lastChild.textContent =
       FeedUtils.stringsSubFolder.GetStringFromName("imapAclCreateRight");
 
@@ -167,8 +175,21 @@ var FeedSubscriptions = {
 
     return dismissDialog;
   },
 
+  onMouseOverLink(event) {
+    const win = FeedSubscriptions;
+    let link = event.target.href;
+    if (event.target.id == "locationValidate") {
+      const feedLocation = document.getElementById("locationValue").value;
+      link += encodeURIComponent(feedLocation);
+    }
+    win.updateStatusItem("statusText", link);
+    event.target.addEventListener("mouseleave", win.clearStatusInfo, {
+      once: true,
+    });
+  },
+
   refreshSubscriptionView(aSelectFolder, aSelectFeedUrl) {
     this.mTree.view = this.mView;
     this.loadSubscriptions();
 
@@ -2509,9 +2530,9 @@ var FeedSubscriptions = {
     if (aEvent.button != 0 || aEvent.target.disabled) {
       return;
     }
 
-    const learnMoreLink = Services.prefs.getCharPref("rss.learnmore.url");
+    const learnMoreLink = aEvent.target.href;
 
     if (this.mMainWin && learnMoreLink.startsWith("http")) {
       this.mMainWin.focus();
       this.mMainWin.openContentTab(learnMoreLink);
@@ -2523,9 +2544,9 @@ var FeedSubscriptions = {
     if (aEvent.button != 0 || aEvent.target.hasAttribute("disabled")) {
       return;
     }
 
-    const validationQuery = Services.prefs.getCharPref("rss.validator.url");
+    const validationQuery = aEvent.target.href;
 
     if (this.mMainWin && validationQuery.startsWith("http")) {
       const tabmail = this.mMainWin.document.getElementById("tabmail");
       if (tabmail) {
diff --git a/mailnews/extensions/newsblog/feed-subscriptions.xhtml b/mailnews/extensions/newsblog/feed-subscriptions.xhtml
--- a/mailnews/extensions/newsblog/feed-subscriptions.xhtml
+++ b/mailnews/extensions/newsblog/feed-subscriptions.xhtml
@@ -80,9 +80,12 @@
       buttonlabelaccept="&button.close.label;"
     >
       <html:div id="contentPane">
         <html:div id="learnMore">
-          <html:a href="" onclick="FeedSubscriptions.openLearnMoreLink(event);"
+          <html:a
+            id="learnMoreLink"
+            href=""
+            onclick="FeedSubscriptions.openLearnMoreLink(event);"
             >&learnMore.label;
           </html:a>
         </html:div>
 
